lib/ansible/modules/network/ios/ios_config.py

```diff
@@ -392,7 +392,13 @@ def get_running_config(module, current_config=None, flags=None):
 def save_config(module, result):
     result['changed'] = True
     if not module.check_mode:
-        run_commands(module, 'copy running-config startup-config\r')
+        try:
+            get_connection(module).send_command(command='copy running-config startup-config\r',
+                                                prompt='Overwrite the previous NVRAM configuration?[confirm]',
```

Comment:
You almost had it right the first time, but you needed to use a raw string instead (or double escape, but this is why we have raw strings)

```suggestion
                                                prompt=r'Overwrite the previous NVRAM configuration\?\[confirm\]',
```